pkg/repositories/gormimpl/project_repo.go

```diff
@@ -70,3 +70,24 @@ func NewProjectRepo(db *gorm.DB, errorTransformer errors.ErrorTransformer,
 		metrics:          metrics,
 	}
 }
+
+func (r *ProjectRepo) UpdateProject(ctx context.Context, project models.Project, projectUpdate models.Project) (error) {
```

Comment:
since we're going with a merge patch type strategy i think you don't even need the existing project model to begin with (http://gorm.io/docs/update.html#Update-Changed-Fields)

Reply:
Question: also for @anandswaminathan and folks who had opinions — do we want to update fields even if the update is null? In that case, we could skip both the if branches in this function as well.